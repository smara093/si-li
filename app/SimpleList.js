import React from "react";
import {
  Text,
  View,
  FlatList,
  TouchableHighlight,
  Button,
  ListView,
  TextInput
} from "react-native";
import { connect } from "react-redux";
import { actionCreators } from "./listRedux";
import styles from "./styles";
import Title from "./Title";

const mapStateToProps = state => ({
  items: state.items,
  newItem: state.newItem
});

class SimpleList extends React.Component {
  removeItem = index => {
    const { dispatch } = this.props;
    dispatch(actionCreators.remove(index));
  };

  renderItem = ({ item, index }) => {
    return (
      <TouchableHighlight
        onLongPress={() => {
          this.removeItem(index);
        }}
      >
        <View style={[styles.row, !item.isActive && styles.inactiveRow]}>
          <Text>{item.text}</Text>
        </View>
      </TouchableHighlight>
    );
  };

  addItem = () => {
    const { dispatch, newItem } = this.props;
    if (newItem) {
      dispatch(
        actionCreators.add({
          key: Date.now(),
          text: newItem,
          lastModified: Date.now(),
          isActive: true
        })
      );
    }
  };

  changeText = text => {
    const { dispatch } = this.props;
    dispatch(actionCreators.updateText(text));
  };

  itemKeyExtractor = ({ key }) => key;

  clearItems = () => {
    const { dispatch } = this.props;
    dispatch(actionCreators.clearItems());
  };

  render() {
    const { items, newItem } = this.props;

    return (
      <View style={styles.container}>
        <Title styles={styles} text="a simple list" />
        <View style={{ height: 60, flexDirection: "row", padding: 10 }}>
          <TextInput
            placeholder="type to add a new item"
            value={newItem}
            onSubmitEditing={this.addItem}
            onChangeText={this.changeText}
            style={{ flex: 1 }}
            underlineColorAndroid="transparent"
          />
          <Button
            title="Add"
            onPress={this.addItem}
            color="purple"
            style={{ flex: 1 }}
          />
        </View>
        <FlatList
          data={items}
          renderItem={this.renderItem}
          keyExtractor={this.itemKeyExtractor}
          extraData={this.state}
        />
        <Button title="Clear Items" onPress={this.clearItems} color="purple" />
      </View>
    );
  }
}

export default connect(mapStateToProps)(SimpleList);
