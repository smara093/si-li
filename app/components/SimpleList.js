import React from 'react';
import { View, FlatList, Button, TextInput } from 'react-native';
import { PropTypes } from 'prop-types';

import styles from './styles/SimpleListStyles';
import SimpleListItem from './SimpleListItem';

const itemKeyExtractor = item => item.id;

class SimpleList extends React.Component {
  // move this on select list action
  componentWillMount() {
    this.props.onComponentInit();
  }

  renderItem(item, index) {
    const { onRemoveItem, onSelectItem } = this.props;

    return (
      <SimpleListItem
        onRemoveItem={onRemoveItem}
        index={index}
        item={item}
        styles={styles}
        onSelectItem={onSelectItem}
      />
    );
  }

  render() {
    const {
      items, newItem, onAddItemClick, onChangeText, onClearItemsClick, owner,
    } = this.props;

    return (
      <View style={styles.container}>
        <View style={{ height: 60, flexDirection: 'row', padding: 10 }}>
          <TextInput
            placeholder="type to add a new item"
            value={newItem}
            onSubmitEditing={() => onAddItemClick(newItem)}
            onChangeText={text => onChangeText(text)}
            style={{ flex: 1 }}
            underlineColorAndroid="transparent"
          />
          <Button
            title="Add"
            onPress={() => onAddItemClick(newItem, owner)}
            color="purple"
            style={{ flex: 1 }}
          />
        </View>
        {items &&
          items.length > 0 && (
            <View>
              <FlatList
                data={items}
                renderItem={({ item, index }) => this.renderItem(item, index)}
                keyExtractor={itemKeyExtractor}
                extraData={this.state}
              />
              <Button title="Clear Items" onPress={() => onClearItemsClick()} color="purple" />
            </View>
          )}
      </View>
    );
  }
}

SimpleList.propTypes = {
  newItem: PropTypes.string.isRequired,
  items: PropTypes.array.isRequired,
  onAddItemClick: PropTypes.func.isRequired,
  onChangeText: PropTypes.func.isRequired,
  onClearItemsClick: PropTypes.func.isRequired,
  onRemoveItem: PropTypes.func.isRequired,
  onComponentInit: PropTypes.func.isRequired,
  onSelectItem: PropTypes.func,
  owner: PropTypes.object,
};

SimpleList.defaultProps = {
  onSelectItem: () => {},
  owner: null,
};

export default SimpleList;
