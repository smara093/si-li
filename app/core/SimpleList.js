export default class SimpleList {
    constructor(items = []) {
        this.items = items;
    }
}