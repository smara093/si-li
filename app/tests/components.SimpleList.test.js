import 'react-native';
import React from 'react';
import SimpleList from '../components/SimpleList';

import renderer from 'react-test-renderer';

// TODO: figure out how to mock root component with store and reducer
// test('renders correctly', () => {
//     const tree = renderer.create(<SimpleList />).toJSON();
//     expect(tree).toMatchSnapshot();
// });

test('true is true', () => {
  expect(true).toBe(true);
});
