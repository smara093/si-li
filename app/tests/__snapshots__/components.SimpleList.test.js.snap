// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginTop": 25,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ef7de7",
        "fontSize": 20,
        "marginBottom": 5,
        "padding": 15,
      }
    }
  >
    a simple list
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 60,
        "padding": 10,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="type to add a new item"
      style={
        Object {
          "flex": 1,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "purple",
              },
            ]
          }
        >
          Add
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": 1,
          "isActive": true,
          "lastModified": 2017-02-01T05:00:00.000Z,
          "text": "item 1",
        },
      ]
    }
    disableVirtualization={false}
    extraData={null}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#76B59B",
                    "marginBottom": 5,
                    "padding": 15,
                  },
                  false,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                item 1
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "purple",
            },
          ]
        }
      >
        Clear Items
      </Text>
    </View>
  </View>
</View>
`;
